interfaces/apparmor/backend.go

```diff
@@ -145,7 +145,7 @@ func (b *Backend) combineSnippets(snapInfo *snap.Info, opts interfaces.Confineme
 
 func addContent(securityTag string, snapInfo *snap.Info, opts interfaces.ConfinementOptions, snippets map[string][][]byte, content map[string]*osutil.FileState) {
 	policy := defaultTemplate
-	if opts.DevMode && !opts.JailMode {
+	if (opts.DevMode || opts.Classic) && !opts.JailMode {
 		policy = attachPattern.ReplaceAll(policy, attachComplain)
```

Comment:
classic shouldn't complain-- you'll get a ton of syslog entries when the snap is legitimately running in classic. Instead please generate a different profile. Eg:

```
$ cat /var/lib/snapd/apparmor/profiles/snap.iam.classic
#include <tunables/global>

@{SNAP_NAME}="iam"
@{SNAP_REVISION}="x1"
@{INSTALL_DIR}="/snap"

# use complain to future-proof if we missed something
profile "snap.iam.classic" (attach_disconnected,complain) {
  # set file rules so that exec() inherit our profile
  / rwkl,
  /** rwlkm,
  /** pix,

  capability,
  change_profile,
  dbus,
  network,
  mount,
  remount,
  umount,
  pivot_root,
  ptrace,
  signal,
  unix,
}
```